src/S3/3d.c

```diff
@@ -517,21 +516,21 @@ int S3Calculate3D(tS3_channel* chan, int pIs_ambient) {
         dist = sqrtf(dist_squared);
     }
     if (pIs_ambient) {
-        v11 = 1.0 - ((chan->position.z - gS3_listener_position_now.z) * (chan->velocity.z - gS3_listener_vel_now.z) + (chan->velocity.y - gS3_listener_vel_now.y) * (chan->position.y - gS3_listener_position_now.y) + (chan->position.x - gS3_listener_position_now.x) * (chan->velocity.x - gS3_listener_vel_now.x)) / dist / flt_531D98;
+        v11 = 1.0f - ((chan->position.z - gS3_listener_position_now.z) * (chan->velocity.z - gS3_listener_vel_now.z) + (chan->velocity.y - gS3_listener_vel_now.y) * (chan->position.y - gS3_listener_position_now.y) + (chan->position.x - gS3_listener_position_now.x) * (chan->velocity.x - gS3_listener_vel_now.x)) / dist / flt_531D98;
         if (v11 <= 2.0f) {
-            if (v11 < 0.5) {
-                v11 = 0.5;
+            if (v11 < 0.5f) {
+                v11 = 0.5f;
             }
         } else {
-            v11 = 2.0;
+            v11 = 2.0f;
         }
```

Comment:
While you're looking at this function:
```suggestion
        if (v11 > 2.0f) {
            v11 = 2.0f;
        else if (v11 < 0.5f) {
            v11 = 0.5;
        }
```

Reply:
yep 👍